Gems/Atom/RHI/Code/Include/Atom/RHI.Reflect/PipelineLibraryData.h

```diff
@@ -53,7 +65,10 @@ namespace AZ
             PipelineLibraryData(AZStd::vector<uint8_t>&& data);
             PipelineLibraryData() = default;
 
-            AZ_SERIALIZE_FRIEND();
+            template <typename, typename>
+            friend struct AnyTypeInfoConcept;
+            template <typename, bool, bool>
+            friend struct Serialize::InstanceFactory;
```

Comment:
I'm not particularly a fan of getting rid of the macro and replacing it with the expanded version. By doing this we lose the comments explaining what it is and why it's here, and it's not as easy to understand how to add it to new classes. I won't block the PR over it, but please at least consider bringing back the macro in a much smaller SerializeContextFriend.h or something.

Reply:
Micro include added as [SerializeContext_fwd.h](https://github.com/aws-lumberyard-dev/o3de/blob/serialize-context-class-builder-updates/Code/Framework/AzCore/AzCore/Serialization/SerializeContext_fwd.h)